configserver/src/main/java/com/yahoo/vespa/config/server/application/CompressedApplicationInputStream.java

```diff
@@ -40,88 +43,71 @@
      */
     public static CompressedApplicationInputStream createFromCompressedStream(InputStream is, String contentType) {
         try {
-            ArchiveInputStream ais = getArchiveInputStream(is, contentType);
-            return createFromCompressedStream(ais);
-        } catch (IOException e) {
+            Options options = Options.standard().allowDotSegment(true);
+            switch (contentType) {
+                case ApplicationApiHandler.APPLICATION_X_GZIP:
+                    return new CompressedApplicationInputStream(ArchiveStreamReader.ofTarGzip(is, options));
+                case ApplicationApiHandler.APPLICATION_ZIP:
+                    return new CompressedApplicationInputStream(ArchiveStreamReader.ofZip(is, options));
+                default:
+                    throw new BadRequestException("Unable to decompress");
+            }
+        } catch (UncheckedIOException e) {
             throw new InternalServerException("Unable to create compressed application stream", e);
         }
     }
 
-    static CompressedApplicationInputStream createFromCompressedStream(ArchiveInputStream ais) {
-        return new CompressedApplicationInputStream(ais);
-    }
-
-    private static ArchiveInputStream getArchiveInputStream(InputStream is, String contentTypeHeader) throws IOException {
-        ArchiveInputStream ais;
-        switch (contentTypeHeader) {
-            case ApplicationApiHandler.APPLICATION_X_GZIP:
-                ais = new TarArchiveInputStream(new GZIPInputStream(is));
-                break;
-            case ApplicationApiHandler.APPLICATION_ZIP:
-                ais = new ZipArchiveInputStream(is);
-                break;
-            default:
-                throw new BadRequestException("Unable to decompress");
-        }
-        return ais;
-    }
-
-    private CompressedApplicationInputStream(ArchiveInputStream ais) {
-        this.ais = ais;
-    }
-
     /**
      * Close this stream.
      * @throws IOException if the stream could not be closed
      */
     public void close() throws IOException {
-        ais.close();
+        reader.close();
     }
 
     File decompress() throws IOException {
```

Comment:
Would be nice if this was cleaned up, rather than just added to .gitignore :)

Reply:
The gitignore change was unrelated to this. CompressedApplicationInputStreamTest doesn't always close though, is that what you mean?

Reply:
Ah, I noticed my /var/ under config-application-package, or something, was filled up after running these unit tests. I assumed that's why you added the ignore, and also because of this test.